src/lib/celebrations.ts

```diff
@@ -0,0 +1,291 @@
+      prioritized: true,
+    },
+    {
+      key: 'epiphany',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.epiphany(year, config.epiphanyOnSunday),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+    },
+    {
+      key: 'trinity_sunday',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.trinitySunday(year),
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+      cycles: { liturgicalDayCycle: LiturgicalDayCycle.TEMPORALE },
+    },
+    {
+      key: 'corpus_christi',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.corpusChristi(year, config.corpusChristiOnSunday),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+      cycles: { liturgicalDayCycle: LiturgicalDayCycle.TEMPORALE },
+    },
+    {
+      key: 'most_sacred_heart_of_jesus',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.mostSacredHeartOfJesus(year),
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+      cycles: { liturgicalDayCycle: LiturgicalDayCycle.TEMPORALE },
+    },
+    {
+      key: 'nativity_of_john_the_baptist',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.nativityOfJohnTheBaptist(year),
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+    },
+    {
+      key: 'peter_and_paul_apostles',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.peterAndPaulApostles(year),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.RED,
+      prioritized: true,
+    },
+    {
+      key: 'assumption',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.assumption(year),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+    },
+    {
+      key: 'all_saints',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.allSaints(year),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+    },
+    {
+      key: 'christ_the_king_sunday',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.christTheKingSunday(year),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+      cycles: { liturgicalDayCycle: LiturgicalDayCycle.TEMPORALE },
+    },
+    {
+      key: 'joseph_spouse_of_mary',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.josephSpouseOfMary(year),
+      isHolyDayOfObligation: true,
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+    },
+    {
+      key: 'annunciation',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.annunciation(year),
+      liturgicalColors: LiturgicalColors.WHITE,
+      prioritized: true,
+    },
+    {
+      key: 'easter_sunday',
+      rank: Ranks.SOLEMNITY,
+      date: Dates.easter(year),
+      isHolyDayOfObligation: true,
+      seasons: [LiturgicalSeasons.PASCHAL_TRIDUUM, LiturgicalSeasons.EASTERTIDE],
```

Comment:
We can set the rule here in `celebration.ts` or in `seasons.ts`. It's the same. But indeed `season` properties are commonly set in `seasons.ts`. I'll fix it.

And Easter Sunday is still part of the Paschal Triduum, until Sunday evening. There is no first Vesper of Easter, Easter Sunday start with the Vigils. So the most part of Easter Sunday is within 2 seasons.

Reply:
Aaah, sorry, I always thought that the Paschal Triduum is three days long, from Thursday to Satuday, however, obviously I am wrong (I’ve just read the [GNLY](https://www.archbalt.org/wp-content/uploads/2017/04/GeneralNormsLiturgicalYear.pdf#page=3)).

---

I don’t want to be rude, but: I see another issue we already talked about: IMHO we should separate celebration definitions of Holy Saturday and the vigil of Easter Sunday.

AFAIK, even in the _Ordos_ and _Roman Missals, they list them separately (i.e. there are two keys  and two celebration definitions, which would break the test of having 365/366 objects in the romcal object when no optional memorial is output). That is:
- one for Holy Saturday;
- one for Holy Saturday evening.

Reasoning:
- Holy Saturday has different liturgical colour than Easter Sunday (although no Mass is served, but the Liturgy of the Hours is prayed and it can be prayed with a deacon/priest/bishop in _pluviale_ (cape), which should be in the liturgical colour of the day;
- Holy Saturday object should contain information about celebration of the Liturgy of the Hours (once implemented, see #20), which is (obviously) different from that of the Easter Sunday.